core/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridParser.java

```diff
@@ -40,46 +36,19 @@
     public static final int DEFAULT_PRECISION = 5;
     public static final int DEFAULT_MAX_NUM_CELLS = 10000;
 
+    private final ObjectParser<GeoGridAggregationBuilder, QueryParseContext> parser;
```

Comment:
I was imagining removing this class entirely and moving the parser to a static function/`public static final ObjectParser` variable in `GeoGridAggregationBuilder` I'd have made the `addFields` method a static method on `AbstractValuesSourceParser` and probably called it something different. Or removed `AbstractValuesSourceParser` entirely and moved the static methods to some other spot. I'm not sure about that. Basically I figured we don't even need this class and should try to get away with something static in the builder.